xchange-poloniex/src/main/java/info/bitrich/xchangestream/poloniex/PoloniexStreamingMarketDataService.java

```diff
@@ -22,10 +22,27 @@
 import java.util.*;
 
 public class PoloniexStreamingMarketDataService implements StreamingMarketDataService {
+    public static final String TICKER_CHANNEL_NAME = "ticker";
     private final WampStreamingService streamingService;
+    private Observable<Ticker> streamingTickers;
 
     public PoloniexStreamingMarketDataService(WampStreamingService streamingService) {
         this.streamingService = streamingService;
+        this.streamingTickers = streamingService.subscribeChannel(TICKER_CHANNEL_NAME)
```

Comment:
Rather than always subscribing to tickers even if they aren't needed, why not do this using `Suppliers.memoise()` from guava? Then the stream is initialised lazily if needed.

Reply:
An outstanding idea. It is done.